src/transaction.ts

```diff
@@ -29,44 +29,45 @@ import {PartialResultStream, partialResultStream, ResumeToken, Row} from './part
 import {Session} from './session';
 import {Key} from './table';
 import {SpannerClient as s} from './v1';
+import {GenericCallback, BasicCallback, ITransactionOptions, ITransaction, ITimestamp, IReadOnly, ICommitResponse, ICommitRequest} from './common';
+
+export type BeginTransactionCallback = GenericCallback<ITransaction>;
+export type CommitCallback = GenericCallback<ICommitResponse>;
+export type ReadCallback = GenericCallback<Rows>;
+export type RunUpdateCallback = GenericCallback<number>;
 
 export type Rows = Array<Row|Json>;
+export type BeginResponse = [s.Transaction];
+export type CommitResponse = [ICommitResponse];
+export type ReadResponse = [Rows];
+export type RunResponse = [Rows, s.ResultSetStats];
+export type RunUpdateResponse = [number];
 
-export interface TimestampBounds {
-  strong?: boolean;
-  minReadTimestamp?: PreciseDate|p.ITimestamp;
-  maxStaleness?: number|p.IDuration;
-  readTimestamp?: PreciseDate|p.ITimestamp;
-  exactStaleness?: number|p.IDuration;
-  returnReadTimestamp?: boolean;
-}
+export type BatchUpdatePromise = Promise<[number[], s.ExecuteBatchDmlResponse]>;
+export type BeginPromise = Promise<[s.Transaction]>;
+export type CommitPromise = Promise<[s.CommitResponse]>;
```

Comment:
I see some mixing between the interfaces declare in the common file and the gapic types. Can we use one or the other?